### PR TITLE
Add CSS Linter and Helper

### PR DESCRIPTION
**Pull request by SMahil**
### Mandatory
 - [x] What does this PR do? 
This PR will
1.  Add CSS Helper/Linter(CSSComb) which will format code as per given set of rules
2. Add/Improve rules for LessHint Linter
3. Fix gulp task to fail to build if linter fails
4. Fix CSS as per linter rules

-------------------------------------------
#### Run below command to fix most of the linting errors
`csscomb src`

##### To Lint specific file, run 
`csscomb "filepath"`

-------------------------------------------
#### Limitations

CSSComb will not lint following linting errors

1. LESS functions
2. will not add attribute quotes automatically
3. hex color values for color variables
4. duplicate property
5. space around a comma
6. space around an operator
7. removing the universal selector

--------------------------------------------

- [ ] Are the tests Included?
NA
___
### Optional
- [ ] Is the documentation Included?

- [ ] @mention(s) to expected reviewer(s) included
@nimishamukherjee @rgarg1

### CODE REVIEW

**Comment by alien-ike** [comment]
### Ike Plugins (test-keeper)

Thank you @SMahil for this contribution!



It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


Your plugin configuration is stored in the [file](https://github.com/fabric8-ui/fabric8-planner/blob/48093fcbf66a24b432dec25937e2929ec7e0765d/.ike-prow/test-keeper.yaml).

File: gulpfile.js [hunk 1]
```diff
@@ -19,9 +19,7 @@ var gulp = require('gulp')
   , less = require('gulp-less')
   , util = require('gulp-util')
 
-  , changed   = require('gulp-changed')
```
Comment (jarifibrahim) [inline gulpfile.js]: Please remove this from package.json also

File: gulpfile.js [hunk 2]
```diff
@@ -19,9 +19,7 @@ var gulp = require('gulp')
...
   , lesshint  = require('gulp-lesshint')
-  , concat    = require('gulp-concat-css')
```
Comment (jarifibrahim) [inline gulpfile.js]: Please remove this from package.json also

File: gulpfile.js [hunk 3]
```diff
@@ -89,17 +87,15 @@ mach.transpileLESS = function (src, debug) {
...
     .pipe(lesshint({
       configPath: './.lesshintrc' // Options
     }))
+    .pipe(less({
+      plugins: [autoprefix]
+    }))
     .pipe(lesshint.reporter()) // Leave empty to use the default, "stylish"
     .pipe(lesshint.failOnError()) // Use this to fail the task on lint errors
     .pipe(srcmaps.init())
-    .pipe(less(opts))
```
Comment (jarifibrahim) [inline gulpfile.js]: Please remove line 87-90. We don't need it.

Comment (SMahil) [reply gulpfile.js]: Those are kept for reference only

Comment (jarifibrahim) [reply gulpfile.js]: It's commented code. We're better off without it :)

File: src/app/components_ngrx/iteration-list-entry/iteration-list-entry.component.less [hunk 4]
```diff
@@ -7,7 +8,9 @@
     padding-right: 10px;
     border-left: 4px solid;
     border-left-color: transparent;
-    .badge { background-color: @color-pf-black; }
+    .badge {
```
Comment (vikram-raj) [inline src/app/components_ngrx/iteration-list-entry/iteration-list-entry.component.less]: Can you please remove this unused. We are not using this ```.badge``` class in iteration-list-entry component.

File: src/app/widgets/typeahead-dropdown/typeahead-dropdown.component.less [hunk 5]
```diff
@@ -58,4 +67,6 @@
     }
   }
 }
-.detail-action-btn { .detail-action-btn; }
+.detail-action-btn {
+  .detail-action-btn;
+}
```
Comment (vikram-raj) [inline src/app/widgets/typeahead-dropdown/typeahead-dropdown.component.less]: What does this code do?

File: .csscomb.json [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+  "space-after-opening-brace": "\n",
+  "space-after-selector-delimiter": "\n",
+  "space-before-selector-delimiter": "",
+  "space-before-closing-brace": "\n",
+  "strip-spaces": true,
+  "tab-size": true,
+  "unitless-zero": true,
+  "verbose": true,
+  "exclude": ["src/assets/stylesheets/shared/_formulas.less"],
+  "csscomb.preset":
```
[the remark this answers is not in the document]
Comment (SMahil) [reply .csscomb.json]: no, they are not same

File: .csscomb.json [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+  "space-before-combinator": "",
+  "space-after-combinator": " ",
+  "space-between-declarations": "\n",
+  "space-before-opening-brace": " ",
+  "space-after-opening-brace": "\n",
+  "space-after-selector-delimiter": "\n",
+  "space-before-selector-delimiter": "",
+  "space-before-closing-brace": "\n",
+  "strip-spaces": true,
+  "tab-size": true,
```
[the remark this answers is not in the document]
Comment (vikram-raj) [reply .csscomb.json]: But number would be more readable and maintainable. So, Can you change these into number.

File: .csscomb.json [hunk 8]
```diff
@@ -0,0 +1,219 @@
+{
+  "always-semicolon": true,
+  "color-case": "lower",
+  "block-indent": "  ",
```
Comment (vikram-raj) [inline .csscomb.json]: Can you please change this to ```"block-indent": 2```.

**Pull request merged**